lib/rom/mapper/attribute_dsl.rb

```diff
@@ -94,6 +95,21 @@ def exclude(*names)
         attributes.delete_if { |attr| names.include?(attr.first) }
       end
 
+      # Perform transformations sequentially
+      #
+      # @example
+      #   dsl = AttributeDSL.new()
+      #
+      #   dsl.step do
+      #     attribute :name
+      #   end
+      #
+      # @api public
+      def step(options=EMPTY_HASH, &block)
+        raise(ArgumentError, "cannot mix outer attributes and steps") unless attributes.empty?
+        steps << new(options, &block)
+      end
```

Comment:
Hmm *I think* this should belong to `Mapper::DSL::ClassMethods` in this case as it's a higher level concept than AttributeDSL stuff